homeassistant/components/notion/binary_sensor.py

```diff
@@ -75,19 +75,21 @@ class NotionBinarySensor(NotionEntity, BinarySensorEntity):
     """Define a Notion sensor."""
 
     @callback
+    @async_guard_against_missing_task
     def _async_update_from_latest_data(self) -> None:
         """Fetch new state data for the sensor."""
-        task = self.coordinator.data["tasks"][self._task_id]
+        task = self.coordinator.data["tasks"][self.task_id]
 
         if "value" in task["status"]:
             self._state = task["status"]["value"]
         elif task["task_type"] == SENSOR_BATTERY:
             self._state = task["status"]["data"]["to_state"]
 
     @property
+    @async_guard_against_missing_task
```

Comment:
We don't call `is_on` if the sensor is unavailable.

Reply:
See below.